lib/input/reader/amqp_1.go

```diff
@@ -231,15 +249,26 @@ func (a *AMQP1) ReadWithContext(ctx context.Context) (types.Message, AsyncAckFn,
 
 	msg.Append(part)
 
-	var done chan struct{}
+	var renewDoneChannel chan struct{}
 	if a.conf.AzureRenewLock {
-		done = a.startRenewJob(amqpMsg)
+		renewDoneChannel = a.startRenewJob(amqpMsg)
 	}
 
+	a.m.RLock()
+	a.pendingMsgs[pendingMsgIndex(amqpMsg)] = &amqp1PendingMsg{
+		msg: msg,
+		renewDoneChannel: renewDoneChannel,
+	}
+	a.m.RUnlock()
+
 	return msg, func(ctx context.Context, res types.Response) error {
-		if done != nil {
-			close(done)
-			done = nil
+		msg, ok := a.popPendingMessage(amqpMsg)
```

Comment:
Do we really have a stack that we pop?

Reply:
We have a dict from which we get and remove a specific element, the same way as in i.e. python dictionary: https://docs.python.org/3/library/stdtypes.html#dict.pop

Reply:
All good.  Just do not call it `pop` which is for a stack.